packages/zoe/src/contracts/newSwap/getCurrentPrice.js

```diff
@@ -183,14 +189,25 @@ export const makeGetCurrentPrice = (
     } else if (isSecondary(brandIn) && isCentral(brandOut)) {
       // We'll add the protocol fee to amountOut to get deltaY. The user will
       // pay deltaY. The Pool gains deltaY, pays amountOut plus protocolFee
-      const preliminaryProtocolFee = multiplyBy(amountOut, protocolFeeRatio);
+
+      // protocolFee is feeRatio * deltaY, and deltaY is (1 + feeRatio) * amountOut,
+      // so we compute protocolFee as feeRatio * (1 + feeRatio) * amountOut
```

Comment:
I think this language is wrong, but depends on how you're using it. I think in the paper, x is always what is given and y is always the payout. So in that case, x's brand is the secondaryBrand, and y's brand is the centralBrand, because the user is giving in the secondaryBrand, and receiving a payout in the centralBrand. So deltaY would not be what the user is paying. The user would pay something like deltaX (not accounting for fees), the pool gains deltaX and pays deltaY,  and then the protocolFee is subtracted from deltaY. 

But, maybe you have a different way of looking at it. How are you choosing what is x and what is y?